docs/cpu-constraints.md

```diff
@@ -0,0 +1,40 @@
+
+## Containers life cycle
+
+When you create a container with a CPU constraint, the runtime hot adds the number of vCPUs required by the container. Similarly, when the container stops, the runtime hot removes these resources.
+
+## Containers without CPU constraint
+
+A container without a CPU constraint uses the default number of vCPUs specified in the configuration file, in the case of Kubernetes PODs, containers without a CPU constraint use and share between them the default number of vCPUs, for example, if `default_vcpus` is equal to 1 and you have 2 containers without CPU constraint and each container tries to consume 100% of vCPU, since your virtual machine has not enough resources to satisfy containers needs, the resources are divided in two parts, 50% of vCPU for each container. If you wish to give access to a greater or lesser portion of vCPUs to a specific container, you can use [docker --cpu-shares][1] or [kubernetes `cpu` requests][2].
+One thing to consider before running containers without CPU constraint is that your containers are not running alone, since your containers run inside a virtual machine, there other processes trying to use the vCPUs too, for example systemd and the Clear Container [agent][5]. In general, we recommend to set `default_vcpus` to 1 to allow non-container processes run on this vCPU and specify a CPU constraint for each container, if your container is already running and it needs more vCPUs, you can add more using [docker update][4].
+
+## Containers with CPU constraint
+
+The runtime calculates the number of vCPUs required by containers with CPU constraint using the following formula: `(quota + (period -1)) / period`, the result determinate the number of vCPU to hot plug into the virtual machine. Once the vCPUs have been hot added, the [agent][5] places the container inside a CPU cgroup, allowing to the container use only its resources assigned.
+
+## Do not waste resources
+
+If you already know the number of vCPUs needed for each container and POD, or just want to run them with the same number of vCPUs, you can specify that number using the `default_vcpus` option in the configuration file, each virtual machine starts with that number of vCPUs, one limitation of this approach is that these vCPUs can not be hot removed later and you might be wasting resources, for example if you set `default_vcpus` to 8 and run only one container with a CPU constrain of 1 vCPUs, you might be wasting 7 vCPUs, since the virtual machine starts with 8 vCPUs and 1 vCPUs is hot added and assigned to the container, non-container processes might be able to use 8 vCPUs, but they use a maximum 1 vCPU, hence 7 vCPUs might not be used.
```

Comment:
Suggested rewrite to break up the sentences: "One limitation of this approach is that these vCPUs cannot be hot removed later and you might be wasting resources. For example, if you set `default_vcpus` to 8 and run only one container with a CPU constrain of 1 vCPUs, you might be wasting 7 vCPUs since the virtual machine starts with 8 vCPUs and 1 vCPUs is hot added and assigned to the container. Non-container processes might be able to use 8 vCPUs but they use a maximum 1 vCPU, hence 7 vCPUs might not be used."